tensorflow/stream_executor/dnn.h

```diff
@@ -667,8 +667,33 @@ class PoolingDescriptor {
   std::vector<int64> strides_;
 };
 
-typedef int64 AlgorithmType;
-constexpr AlgorithmType kDefaultAlgorithm = -1;
+class AlgorithmType {
+  private:
+  int64 algo_;
+  bool tensor_ops_enabled_;
+  enum { kDefaultAlgorithm = -1 };
+
+  public:
+  AlgorithmType() : algo_(kDefaultAlgorithm), tensor_ops_enabled_(false) { }
+  AlgorithmType(int64 a) : algo_(a), tensor_ops_enabled_(false) { }
+  bool is_default() const { return algo_ == kDefaultAlgorithm; }
+  bool operator==(const AlgorithmType& other) const {
+    return this->algo_ == other.algo_ &&
+           this->tensor_ops_enabled_ == other.tensor_ops_enabled_;
+  }
+  bool tensor_ops_enabled() const {
+    return tensor_ops_enabled_;
+  }
+  void set_tensor_ops(bool b) {
```

Comment:
Why do we have this function instead of a constructor that takes both arguments?

Reply:
Replaced setters with constructor.